### PR TITLE
improve geospatial feature type update

### PR DESCRIPTION
**Pull request by tuanng-cognite**

### CODE REVIEW

File: cognite/client/data_classes/geospatial.py [hunk 1]
```diff
@@ -39,14 +40,18 @@ class FeatureTypeList(CogniteResourceList):
     _ASSERT_CLASSES = False
 
 
-class PropertyAndSearchSpec:
```
Comment (vincent-cognite) [inline cognite/client/data_classes/geospatial.py:42]: this will break backward compatibility...
I think we still need to support the old way.

Comment (tuanng-cognite) [reply cognite/client/data_classes/geospatial.py]: I'll revert the change. The only way I see that can keep backward compatibility is to introduce  new method.

File: cognite/client/_api/geospatial.py [hunk 2]
```diff
@@ -184,6 +184,15 @@ def patch_feature_types(self, patch: Union[FeatureTypePatch, List[FeatureTypePat
                 ...         property_updates=Updates(add={"altitude": {"type": "DOUBLE"}}),
                 ...         search_spec_updates=Updates(add={"altitude_idx": {"properties": ["altitude"]}})
                 ... ))
+
+            Add an additional index to an existing property
+
+                >>> from cognite.client.data_classes.geospatial import Updates
+                >>> from cognite.client import CogniteClient
+                >>> c = CogniteClient()
+                >>> res = c.geospatial.patch_feature_types(update=FeatureTypePatch(external_id="wells",
```
Comment (vincent-cognite) [inline cognite/client/_api/geospatial.py:193]: maybe redo indentation to make the example more readable

File: cognite/client/_api/geospatial.py [hunk 3]
```diff
@@ -160,6 +163,51 @@ def mapper(it):
...
+        """`Patch feature types`
+        <https://docs.cognite.com/api/v1/#operation/updateFeatureTypes>
+
+        Args:
+            patch (Union[FeatureTypePatch, List[FeatureTypePatch]]): the patch to apply
+
+        Returns:
+            FeatureTypeList: The patched feature types.
+
+        Examples:
```
Comment (vincent-cognite) [inline cognite/client/_api/geospatial.py:176]: While you are at it, maybe add an example of multi property index, I think that is not obvious that we can do it.

File: cognite/client/_api/geospatial.py [hunk 4]
```diff
@@ -160,6 +163,51 @@ def mapper(it):
...
+
+            Add one property to a feature type and add index on that property:
+
+                >>> from cognite.client.data_classes.geospatial import Updates
+                >>> from cognite.client import CogniteClient
+                >>> c = CogniteClient()
+                >>> res = c.geospatial.patch_feature_types(update=FeatureTypePatch(external_id="wells",
```
Comment (vincent-cognite) [inline cognite/client/_api/geospatial.py:183]: should be patch

File: cognite/client/data_classes/geospatial.py [hunk 5]
```diff
@@ -65,6 +66,19 @@ def __init__(
...
+@dataclasses.dataclass
+class Updates:
+    add: Dict[str, Any] = None
+    remove: List[str] = None
+
+
+@dataclasses.dataclass
+class FeatureTypePatch:
+    external_id: str = None
+    property_updates: Updates = None
```
Comment (vincent-cognite) [inline cognite/client/data_classes/geospatial.py:78]: should we be consistent and use the parameter name "property_patches" ?

**Comment by vincent-cognite** [comment]
should mark the "old" method deprecated (@deprecation ?)

**Review by vincent-cognite — APPROVED** [review]

**Review by dmivankov — APPROVED** [review]

**Pull request merged**